abnf/odata-aggregation-testcases.yaml

```
@@ -865,11 +833,21 @@ TestCases:
     Input: $apply=outerjoin(Sales as TotalSales,aggregate(Amount with sum as
       Total))/groupby((Name,TotalSales/Total))
 
-  - Name: transformation sequences - sub-hierarchy selection
-    Rule: queryOptions
-    Input: $apply=descendants(SalesOrgHierarchy,filter(Name eq 'US'),keep
-      start)/ancestors(SalesOrgHierarchy,filter(contains(Name,'East')),keep
-      start)/traverse(SalesOrgHierarchy,preorder)
+  - Name: transformation sequences - sub-hierarchy selection (example 40)
+    Rule: queryOptions
+    Input: $apply=filter(ID in $these/$filter(Name eq 'US')/Aggregation.descendants(
+      NodePropertyPath='ID',
+      HierarchyNodes=@nodes,
+      HierarchyQualifier='SalesOrgHierarchy',
+      IncludeSelf=true))/filter(ID in $these/$filter(contains(Name,'East'))/Aggregation.ancestors(
+      NodePropertyPath='ID',
+      HierarchyNodes=@nodes,
+      HierarchyQualifier='SalesOrgHierarchy',
+      IncludeSelf=true))/orderby($these/Aggregation.preorder(
+      NodePropertyPath='ID',
+      HierarchyNodes=@nodes,
+      HierarchyQualifier='SalesOrgHierarchy',
+      Node=@id))&@nodes=$root/SalesOrganizations&@id=ID
```

```suggestion
      IncludeSelf=true))/orderby($these/Aggregation.preorder(
      NodePropertyPath='ID',
      HierarchyNodes=@nodes,
      HierarchyQualifier='SalesOrgHierarchy',
      Node=ID))&@nodes=$root/SalesOrganizations
```
I would prefer `Node=ID`, whereas ABNF demands that the value of the `Node` function parameter be a `parameterAlias`. But [OData-URL, 5.3](https://docs.oasis-open.org/odata/odata/v4.01/os/part2-url-conventions/odata-v4.01-os-part2-url-conventions.html#sec_ParameterAliases) says that parameter aliases replace _literal values_, and `@id=ID` cannot be resolved to a literal value, because `ID` is undefined outside of the `orderby` transformation.